src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -309,41 +303,52 @@ public void setPeriod(String period) {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (containsSummaryTag(ast)) {
-            validateSummaryTag(ast);
-        }
-        else if (!startsWithInheritDoc(ast)) {
-            final String summaryDoc = getSummarySentence(ast);
-            if (summaryDoc.isEmpty()) {
-                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
-            }
-            else if (!period.isEmpty()) {
-                final String firstSentence = getFirstSentence(ast);
-                final int endOfSentence = firstSentence.lastIndexOf(period);
-                if (!summaryDoc.contains(period)) {
-                    log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+        final DetailNode inlineSummaryTag = getInlineSummaryTag(ast);
+        if (inlineSummaryTag == null) {
+            if (!startsWithInheritDoc(ast)) {
+                final String summaryDoc = getSummarySentence(ast);
+                if (summaryDoc.isEmpty()) {
+                    log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
                 }
-                if (endOfSentence != -1
-                        && containsForbiddenFragment(firstSentence.substring(0, endOfSentence))) {
-                    log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+                else if (!period.isEmpty()) {
+                    final String firstSentence = getFirstSentence(ast);
+                    final int endOfSentence = firstSentence.lastIndexOf(period);
+                    if (!summaryDoc.contains(period)) {
+                        log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+                    }
+                    if (endOfSentence != -1
+                            && containsForbiddenFragment(
+                            firstSentence.substring(0, endOfSentence))) {
+                        log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+                    }
                 }
             }
         }
+        else {
+            validateSummaryTag(inlineSummaryTag);
+        }
     }
 
     /**
-     * Checks if summary tag present.
+     * Gets the inline summary tag.
      *
      * @param javadoc javadoc root node.
-     * @return {@code true} if first sentence contains @summary tag.
+     * @return inline summary tag node if present otherwise null
      */
-    private static boolean containsSummaryTag(DetailNode javadoc) {
+    private static DetailNode getInlineSummaryTag(DetailNode javadoc) {
         final Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
                 .filter(SummaryJavadocCheck::isInlineTagPresent)
                 .findFirst()
                 .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);
 
-        return node.isPresent() && isSummaryTag(node.get());
+        final DetailNode result;
+        if (node.isPresent() && isSummaryTag(node.get())) {
```

Comment:
3 parter. Part 3.

Then change this to `if (isPresent && !isSummary) node = Optional.empty();`

Reply:
done